airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py

```diff
@@ -104,11 +104,21 @@ def _field_to_datetime(value: Union[int, str]) -> pendulum.datetime:
 
 
 class TimeIncrementalStream(BaseIncrementalStream, ABC):
-    def __init__(self, start_date, **kwargs):
+    state_checkpoint_interval = 1000
```

Comment:
@RobertoBonnet I believe this is the only line necessary in the current PR. Since this connector wasn't checkpointing previously, a halfway failure causes all progress to be lost. But with this line, the connector will "save" its progress every 1000 records. in that case even if a halfway failure occurs, then we don't lose data.

Reply:
@sherifnada @marcosmarxm It makes sense. Increasing the limit of records that we will process for each request along with the checkpoints will help a lot. Could I do that?

Reply:
is it also possible to use this no IdBased incremental streams? The docs don't mention anything on whether those records are returned in ascending ID order as far as I can see. But it seems really surprising if they don't. 

If records are returned in Ascending order of the ID, we may be able to achieve the same effect by putting this line in `BaseIncrementalStream`

Reply:
@sherifnada These routes that I changed are based on the record update date